### PR TITLE
[TACHYON-1026] Add test coverage for InodeTree's journaling

### PR DESCRIPTION
**Pull request by akshaypundle**
https://tachyon.atlassian.net/browse/TACHYON-1026

 - added hashCode and equals to
   * InodeDirectoryEntry
   * InodeEntry
   * InodeTree

 - added tests

### CODE REVIEW

File: servers/src/main/java/tachyon/master/file/journal/InodeDirectoryEntry.java [hunk 1]
```diff
@@ -49,4 +51,21 @@ public JournalEntryType getType() {
...
+  }
+
+  @Override
+  public boolean equals(Object object) {
+    if (object instanceof InodeDirectoryEntry) {
+      if (!super.equals(object)) {
+        return false;
+      }
+      InodeDirectoryEntry that = (InodeDirectoryEntry) object;
+      return Objects.equal(this.mChildrenIds, that.mChildrenIds);
```
Comment (apc999) [inline servers/src/main/java/tachyon/master/file/journal/InodeDirectoryEntry.java]: In Tachyon codebase, we don't use `this`. Like in this case, we directly use `mChildrenIds`

File: servers/src/main/java/tachyon/master/file/journal/InodeEntry.java [hunk 2]
```diff
@@ -54,4 +55,22 @@ public InodeEntry(long creationTimeMs, long id, String name, long parentId, bool
...
+  public int hashCode() {
+    return Objects.hashCode(mId, mParentId, mName, mIsPinned, mCreationTimeMs,
+        mLastModificationTimeMs);
+  }
+
+  @Override
+  public boolean equals(Object object) {
+    if (object instanceof InodeEntry) {
+      InodeEntry that = (InodeEntry) object;
+      return Objects.equal(this.mId, that.mId) && Objects.equal(this.mParentId, that.mParentId)
```
Comment (apc999) [inline servers/src/main/java/tachyon/master/file/journal/InodeEntry.java]: Same comments apply about `this`

File: servers/src/test/java/tachyon/master/file/meta/InodeTreeTest.java [hunk 3]
```diff
@@ -322,4 +328,78 @@ public void setPinnedTest() throws Exception {
...
+      Mockito.verify(mockOutputStream).writeEntry(node.toJournalEntry());
+    }
+    Mockito.verifyNoMoreInteractions(mockOutputStream);
+  }
+
+  // verify that the tree has the given children
+  private static void verifyChildrenNames(InodeTree tree,InodeDirectory root, Set<String> childNames) throws Exception {
+    List<Inode> children = tree.getInodeChildrenRecursive(root);
+    Assert.assertEquals(childNames.size(), children.size());
+    for(Inode child: children) {
```
Comment (apc999) [inline servers/src/test/java/tachyon/master/file/meta/InodeTreeTest.java]: a space between `for` and `(Inode child: children)` please

**Comment by apc999** [comment]
@akshaypundle, I left a few comments. Thanks!

**Comment by gpang** [comment]
ok to test

**Comment by akshaypundle** [comment]
Thanks for the comments, I have addressed them!

File: servers/src/main/java/tachyon/master/file/journal/InodeEntry.java (hunk 2 above)
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/journal/InodeEntry.java]: @akshaypundle , can you also address this comment?

**Comment by yupeng9** [comment]
LGTM

**Comment by yupeng9** [comment]
@akshaypundle , thanks for the test coverage improvement!

**Pull request merged**